### PR TITLE
etcd: fs: Combine etcd FsCopy tests and add some new test cases

### PR DESCRIPTION
**Pull request by jonathangold**
This is for merge only after #424 as one test will definitely fail until that's fixed. Happy to add some more test cases if anyone has any ideas.
/cc @keur

### CODE REVIEW

**Comment by jonathangold** [comment]
@purpleidea should this commit message be failing the test? It looks fine to me. Bug in regex?

File: etcd/fs/fs_test.go [hunk 1]
```diff
@@ -183,123 +183,81 @@ func TestFs1(t *testing.T) {
 	}
 }
 
-func TestFs2(t *testing.T) {
```
Comment (purpleidea) [inline etcd/fs/fs_test.go]: Why are you nuking these??

Comment (jonathangold) [reply etcd/fs/fs_test.go]: The two tests are identical, and we needed more cases, so I combined them into a single test, and added two more test cases. TestEtcdCopyFs, does the same tests as TestFs2 and TestFs3, but adds one case for a recursive copy and one extra deep copy.

Comment (purpleidea) [reply etcd/fs/fs_test.go]: I've been to lax on this, and judging by my recent commit, I'm obviously not being a good example, so this is a good time to start: if you'd like to make all of those changes, that's fine, but they need to be each done in separate commits (and reviewed individually).

Comment (jonathangold) [reply etcd/fs/fs_test.go]: @purpleidea: I split up the commit, so I add the new test in one, and replace TestFs2 and TestFs3 in the second one.

File: etcd/fs/fs_test.go [hunk 2]
```diff
@@ -306,3 +306,73 @@ func TestFs3(t *testing.T) {
...
+		defer etcdClient.Destroy()
+
+		etcdFs := &etcdfs.Fs{
+			Client:     etcdClient.GetClient(),
+			Metadata:   superblock,
+			DataPrefix: etcdfs.DefaultDataPrefix,
+		}
+
+		if err := etcdFs.MkdirAll(tt.mkdir, umask); err != nil {
+			t.Errorf("mkdir error: %+v", err)
```
Comment (purpleidea) [inline etcd/fs/fs_test.go]: missing a return?

File: etcd/fs/fs_test.go [hunk 3]
```diff
@@ -306,3 +306,73 @@ func TestFs3(t *testing.T) {
...
+		}
+		tree, err := util.FsTree(etcdFs, "/")
+		if err != nil {
+			t.Errorf("tree error: %+v", err)
+			return
+		}
+		t.Logf("tree: \n%s", tree)
+
+		var memFs = afero.NewMemMapFs()
+
```
Comment (purpleidea) [inline etcd/fs/fs_test.go]: nit: kill nl please

File: etcd/fs/fs_test.go [hunk 4]
```diff
@@ -306,3 +306,73 @@ func TestFs3(t *testing.T) {
 		return
 	}
 }
+
+func TestEtcdCopyFs(t *testing.T) {
```
Comment (purpleidea) [inline etcd/fs/fs_test.go]: Let's give this an integer at the end. 0 or 1 or whatever is free :)

**Comment by purpleidea** [comment]
AFAICT in the terrible GitHub UI, the second commit only deletes stuff since you've already added it in the first test, correct? If so, let's just leave those stragglers in. yay, one less commit to write or rebase :)

unless I'm missing something... Thanks for the patch!

**Comment by jonathangold** [comment]
@purpleidea I mixed up my git history. It's all sorted now. Made a few changes, and added back the missing tests.

**Comment by purpleidea** [comment]
LGTM :)

**Comment by purpleidea** [comment]
Now merged, sorry for the delay.

**Pull request merged**